### DISCUSSION TITLE
Slow Update Interval

### DISCUSSION THREAD

**Discussion by funks81** [wills106/homeassistant-solax-modbus#543] [Q&A]
I installed the integration primarily so I could automatically change my battery overnight charge settings from Home Assistant.  I have the polling frequency set at the default of 15 seconds - but have recently (last few weeks - even prior to the latest update) found that my update interval is often 3-4 minutes and sometimes more.  Could anyone point me in the right direction of how to troubleshoot what might be causing this delay in data appearing in home assistant.

**Comment by wills106** [comment]
What:
Version of the Integration?
Version of HA?
Inverter do you have?

How exactly are you connecting to the Inverter?

Do you get any errors in your log?

**Comment by funks81** [comment]
There are no updates available - so presume latest version of the integration - but can’t see where this is shown - as I put in my original post. Again - latest version of home assistant - 2023.9.3Solis solis rhi-6k-48es-5gConnected via Modbus TCPFollowing errors:Logger: homeassistant.components.sensorSource: helpers/entity_platform.py:857 Integration: Sensor (documentation, issues) First occurred: 11:26:20 (57 occurrences) Last logged: 19:25:47Updating solax_modbus sensor took longer than the scheduled update interval 0:00:30Updating octopus_energy sensor took longer than the scheduled update interval 0:00:30Updating sun sensor took longer than the scheduled update interval 0:00:30Sent from my iPhoneOn 29 Sep 2023, at 18:35, wills106 ***@***.***> wrote:﻿
What:
Version of the Integration?
Version of HA?
Inverter do you have?
How exactly are you connecting to the Inverter?
Do you get any errors in your log?
